### PR TITLE
Make JSONDecodable handle String->Double conversion too

### LINKED ISSUE #204: Freddy fails to decode large integers on 32-bit devices
If I attempt to decode a large number on a 32-bit device, such as an iPhone 5, I get errors like:

```
ValueNotConvertible(1472861857112, Swift.Double)
```

Seems to be because the number overflows, triggers a `InternalError.NumberOverflow` in the parser, and `JSONDecodable` cannot convert a `String` to a `Double`. I think there are a few workarounds:

1. Make `JSONDecodable`'s `Double` extension handle `String` objects.
2. Force `JSONParser` to use `Int64` for decoding integral values, always.
3. Attempt to decode with `Int64` on 32-bit architectures if `Int` decoding overflowed.

### PR DESCRIPTION
**Pull request by cbrauchli**
Fixes #204.

To test, run the the MobileFreddyTests on a 32-bit simulator.

### CODE REVIEW

File: Sources/JSONDecodable.swift [hunk 1]
```diff
@@ -32,6 +32,12 @@ extension Double: JSONDecodable {
             self = double
         case let .Int(int):
             self = Swift.Double(int)
+        case let .String(str):
+            if let double = Swift.Double(str) {
+                self = double
+            } else {
+                throw JSON.Error.ValueNotConvertible(value: json, to: Swift.Double)
```
Comment (mdmathias) [inline Sources/JSONDecodable.swift]: The stuttering on `throw JSON.Error.ValueNotConvertible(value: json, to: Swift.Double)` bothers me for some reason. Can we rewrite this to be:

```swift
    public init(json: JSON) throws {
        switch json {
        case let .Double(double):
            self = double
        case let .Int(int):
            self = Swift.Double(int)
       case let .String(str) where Swift.Double(str) != nil:
            self = Swift.Double(str)!
        default:
            throw JSON.Error.ValueNotConvertible(value: json, to: Swift.Double)
        }
    }
```

Comment (cbrauchli) [inline Sources/JSONDecodable.swift]: Sure! Much nicer.

**Comment by mdmathias** [comment]
This looks fine to me. It needs to be updated with master before it can be merged.

Also, can we consolidate this branch with [the work](https://github.com/bignerdranch/Freddy/pull/194/files_) here?

**Comment by alexsporn** [comment]
I added @cbrauchli tests to #194 and updated accordingly

**Comment by cbrauchli** [comment]
@alexsporn is doing this and more in #194.

**Pull request closed without merging**